Add known issues around transit managed keys (#23080)
hashicorp/vault: website/content/partials/known-issues/transit-managed-keys-sign-fails.mdx

### Transit Sign API calls with managed keys fail

#### Affected versions

> - 1.14.0+ up to 1.14.3 inclusively.

Reviewer: ```suggestion
- 1.14.0+ up to 1.14.3 inclusively
```

Style correction: format similar page elements consistently

Author: Applied in https://github.com/hashicorp/vault/pull/23080/commits/73a206f089239be8eee21737a8db955c6bbe762b

#### Issue

> The Transit sign API call responds with an error "requested version for signing does not contain a private part" when
> a Transit sign request uses a managed key.

Reviewer: ```suggestion
Vault responds to Transit sign API calls with the following error when the request uses a managed key:

`requested version for signing does not contain a private part`
```

Style correction: Make key information easy to scan for, make it clear who/what is taking action

Author: Applied in https://github.com/hashicorp/vault/pull/23080/commits/73a206f089239be8eee21737a8db955c6bbe762b and updated missing newline within https://github.com/hashicorp/vault/pull/23080/commits/8672872bc96e136c707c01da8d32cc08e0f97c2f

> <Note>Signing APIs are not affected with non-managed key, key types</Note>

Reviewer 2: ```suggestion
<Note>Signing APIs are not affected with transit native keys</Note>
```

perhaps?

Reviewer 2: Just trying to differentiate the two clearly :-)

Author: Good suggestion, thanks!

> <Note>Signing APIs are not affected with non-managed key, key types</Note>

Reviewer: ```suggestion
<Note>

The issue does not affect signing with the following key types:

- non-managed keys

</Note>
```

Style correction: make key information easy to scan

Author: Applied in https://github.com/hashicorp/vault/pull/23080/commits/73a206f089239be8eee21737a8db955c6bbe762b, just tweaked non-managed keys to Transit native keys
